sycl/test/regression/macro_conflict.cpp

```diff
@@ -0,0 +1,12 @@
+// RUN: %clangxx -fsyntax-only -Xclang -verify %s -o %t.out
+// expected-no-diagnostics
+//
+//===----------------------------------------------------------------------===//
+// This test checks if the user-defined macros SUCCESS is
+// conflicting with the symbols defined in SYCL header files.
+// This test only checks compilation errorr, so the main function is omitted.
```

Comment:
```suggestion
// This test only checks compilation error, so the main function is omitted.
```

Reply:
Thanks, Ronan. I fixed the typo.

Reply:
Thanks.
But can you rename the enumerators according to the coding standard too, since you are working on it?

Reply:
The coding standard that you linked says that enum variables should start with an upper-case letter and does not say anything for the rest. It also mentioned about adding some prefix.
"SYCL_ENQUEUE_SUCCESS" is not following this guideline?
If so, how about the following?
```
  enum ResultT {
    SyclEnqueueSuccess,
    SyclEnqueueBlocked,
    SyclEnqueueFailed
  };
```
I do not like "R_Success". This is not as intuitive as "SYCL_ENQUEUE_SUCCESS".

Reply:
Yes that seems better.
Not having full uppercase limits the risk of macros collisions because often coding rules promote full uppercase for macros...

Reply:
@keryell, I fixed the enum variables as we discussed.
I see the tests: buildbot/sycl-ubu-x64-pr and buildbot/sycl-win-x64 failed.
Are these failures caused by my code changes or existing failures?

Reply:
> The coding standard that you linked says that enum variables should start with an upper-case letter and does not say anything for the rest.

You are right, but the itemize list starts also with a general statement "In general, names should be in camel case (e.g. `TextFileReader` and `isLValue()`). So I understand this as the rule to follow and the first letter as uppercase adds to this general case, instead of replacing it.